### PR TITLE
add rancher upload artifacts cmd

### PR DESCRIPTION
**Pull request by tashima42**
Add a command to download GitHub artifacts and upload them to the prime-artifacts bucket, replacing the registry info

```bash
release upload rancher artifacts v2.8.3
```
config:
```json
     "v2.8.3": {
       "release_branch": "release/v2.8",
       "dry_run": false,
       "skip_status_check": true,
       "rancher_repo_owner": "rancher",
       "base_registry": "docker.io",
       "registry": "registry.rancher.com",
       "prime_artifacts_bucket": "prime-artifacts"
     },

```

### CODE REVIEW

File: cmd/release/cmd/upload.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/rancher/ecm-distro-tools/release/rancher"
+	"github.com/rancher/ecm-distro-tools/repository"
+
```
Comment (briandowns) [inline cmd/release/cmd/upload.go:12]: Remove new line. Let's keep 2 groups. 1 for stdlib and the other for 3rd party libs.

File: cmd/release/cmd/upload.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+var uploadRancherArtifactsCmd = &cobra.Command{
+	Use:   "artifacts [version]",
+	Short: "Upload rancher artifacts from a GitHub Release to an s3 bucket",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return errors.New("expected at least one argument: [version]")
+		}
+		version := args[0]
+		rancherRelease, found := rootConfig.Rancher.Versions[version]
```
Comment (briandowns) [inline cmd/release/cmd/upload.go:34]: The idiom is `ok` for the returned bool.

File: release/rancher/rancher.go [hunk 3]
```diff
@@ -482,6 +487,59 @@ func readStringChan(ch <-chan string) []string {
...
+		return errors.New("the tag isn't a valid semver: " + releaseTag)
+	}
+	fmt.Println("getting release by tag: " + releaseTag)
+	release, _, err := ghClient.Repositories.GetReleaseByTag(ctx, rancherRelease.RancherRepoOwner, rancherRepo, releaseTag)
+	if err != nil {
+		return errors.New("failed to get release by tag: " + err.Error())
+	}
+	httpClient := ecmHTTP.NewClient(time.Second * 15)
+	releaseAssets := make(map[string][]byte)
+	fmt.Println("donwloading release assets")
```
Comment (rafaelbreno) [inline release/rancher/rancher.go:502]: typo `download`

**Comment by tashima42** [comment]
Updated

File: cmd/release/cmd/upload.go (hunk 1 above)
Comment (rafaelbreno) [reply cmd/release/cmd/upload.go:9]: following the structure that @briandowns mentioned, shouldn't we just use `goimports` or `gofmt` to handle the formatting?

**Review by rafaelbreno — APPROVED** [review]
left a couple comments, besides that lgtm.

File: cmd/release/cmd/upload.go (hunk 1 above)
Comment (tashima42) [reply cmd/release/cmd/upload.go:9]: `goimports` doesn't work properly if there is a name conflict, so I had to manually add the packages to the file

**Review by briandowns — APPROVED** [review]

**Review by rafaelbreno — APPROVED** [review]

**Pull request merged**